Mathlib/CategoryTheory/Enriched/Limits/HasConicalLimits.lean

```diff
@@ -0,0 +1,128 @@
+- `HasConicalLimitsOfShape J`: All functors `F : J ⥤ C` have conical limits.
+- `HasConicalLimitsOfSize.{v₁, u₁}`: For all small `J` all functors `F : J ⥤ C` have conical limits.
+- `HasConicalLimits `: `C` has all (small) conical limits.
+
+## References
+
+* [Kelly G.M., *Basic concepts of enriched category theory*][kelly2005]:
+  See section 3.8 for a similar treatment, although the content of this file is not directly
+  adapted from there.
+-/
+
+universe v₁ u₁ v₂ u₂ w v' v u u'
+
+namespace CategoryTheory.Enriched
+
+open Limits
+
+section Definitions
+
+-- note: for the classes it seems that instance inference wants `V` to be an `outParam`.
+variable {J : Type u₁} [Category.{v₁} J]
+variable (V : outParam <| Type u') [Category.{v'} V] [MonoidalCategory V]
+variable (C : Type u) [Category.{v} C] [EnrichedOrdinaryCategory V C]
+
+variable {C} in
+
+/--
+`HasConicalLimit F` represents the mere existence of a conical limit for `F`.
+-/
+class HasConicalLimit (F : J ⥤ C) : Prop extends HasLimit F where
+  preservesLimit_eCoyoneda (X : C) : PreservesLimit F (eCoyoneda V X) := by infer_instance
+
+attribute [instance] HasConicalLimit.preservesLimit_eCoyoneda
+
+variable (J) in
+
+/--
+`C` has conical limits of shape `J` if there exists a conical limit for every functor `F : J ⥤ C`.
+-/
+class HasConicalLimitsOfShape : Prop where
+  /-- All functors `F : J ⥤ C` from `J` have limits. -/
+  hasConicalLimit : ∀ F : J ⥤ C, HasConicalLimit V F := by infer_instance
+
+attribute [instance] HasConicalLimitsOfShape.hasConicalLimit
+
+/--
+`C` has all conical limits of size `v₁ u₁` (`HasLimitsOfSize.{v₁ u₁} C`)
+if it has conical limits of every shape `J : Type u₁` with `[Category.{v₁} J]`.
+-/
+@[pp_with_univ]
+class HasConicalLimitsOfSize : Prop where
+  /-- All functors `F : J ⥤ C` from all small `J` have conical limits -/
+  hasConicalLimitsOfShape : ∀ (J : Type u₁) [Category.{v₁} J], HasConicalLimitsOfShape J V C := by
+    infer_instance
+
+attribute [instance] HasConicalLimitsOfSize.hasConicalLimitsOfShape
+
+/-- `C` has all (small) conical limits if it has limits of every shape that is as big as its
+hom-sets. -/
+abbrev HasConicalLimits : Prop := HasConicalLimitsOfSize.{v, v} V C
+
+end Definitions
+
+section Results
+
+variable {J : Type u₁} [Category.{v₁} J] {J' : Type u₂} [Category.{v₂} J']
+variable (V : Type u') [Category.{v'} V] [MonoidalCategory V]
+variable (C : Type u) [Category.{v} C] [EnrichedOrdinaryCategory V C]
+namespace HasConicalLimit
+
+variable {C} {F G : J ⥤ C} [HasConicalLimit V F]
+
+/-- ensure existence of a conical limit implies existence of a limit -/
+example : HasLimit F := inferInstance
+
+end HasConicalLimit
+
+namespace HasConicalLimitsOfShape
+
+variable [HasConicalLimitsOfShape J V C]
+
+/-- existence of conical limits (of shape) implies existence of limits (of shape) -/
+instance instHasLimitsOfShape : HasLimitsOfShape J C where
+  has_limit _ := inferInstance
+
+end HasConicalLimitsOfShape
```

Comment:
These blocks are not super-reabable, could write these in a more compact way as:
```
instance [HasConicalLimitsOfShape J V C] : HasLimitsOfShape J C where
  has_limit _ := inferInstance
```

Reply:
The point of these blocks is that more lemmas will be added as the API grows. But, sure, I can do that and reintroduce them later when relevant.

Reply:
done